assets/src/data/model/primary-keys.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External imports
+ */
+import { data, Exception } from '@eventespresso/eejs';
+import { __ } from '@eventespresso/i18n';
+import { isArray, reduce, trimEnd, keyBy, isEmpty } from 'lodash';
+import memoize from 'memize';
+
+/**
+ * Internal imports
+ */
+import { validateEntityHasKey } from './validators';
+
+/**
+ * Exposes a map of modelname to primary key exposed by the eejs.data global
+ * via the server.
+ *
+ * @type {{}}
+ */
+export const { primary_keys: primaryKeys = {} } = data.paths;
+
+/**
+ * Returns the values for the given keys from the provided entity.
+ * This function would be used for models that have combined primary keys
+ * (delivered as an array).
+ *
+ * @type { memoized }
+ * @return { string } The string representation for the values.
+ * @throws { Exception }
+ */
+export const valuesForCombinedPrimaryKeys = memoize( ( keys, entity ) => {
+	if ( ! isArray( keys ) ) {
+		throw new Exception(
+			__( 'The provided value is not an array.', 'event_espresso' ),
+		);
+	}
```

Comment:
`! isArray()` is used in at least a couple of places so I wonder if it would be a good candidate for another function in `validators.js`?

Reply:
hmm good point. Will do.